main/views.py

```diff
@@ -2,11 +2,14 @@
 
 import logging
 
+from django.contrib.auth.mixins import LoginRequiredMixin
 from django.http import HttpRequest, HttpResponse
 from django.shortcuts import render
-from django.views.generic.edit import FormView
+from django.urls import reverse
+from django.views.generic.edit import FormView, UpdateView
 
 from .forms import CustomUserCreationForm
+from .models import User
```

Comment:
This should be using the `get_user_model` function. Like what is done in #403 

That also might solve the mypy issues if you're able to copy the structure there with defining a `UserType`